app.js

```diff
@@ -0,0 +1,187 @@
+//실제 todo list 내용들이 저장되는 배열
+const todoList = [];
+
+const $todoInput = document.querySelector("#new-todo-title"); // 할 일 입력칸인 input태그 요소
+const $todoList = document.querySelector("#todo-list"); // todo list 목록들이 담길 ul태그 요소
+const $todoCount = document.querySelector(".todo-count"); // 총 ?개를 나타내는 span태그 요소
+
+const $viewAllList = document.querySelector('.all'); // '전체보기' 버튼
+const $viewTodoList = document.querySelector('.active'); //'해야할 일' 버튼
+const $viewDoneList = document.querySelector('.completed'); //'완료한 일' 버튼
+
+todoListRender();
+
+function todoListRender() {
+    // todoList 배열에 담긴 목록 정보를 ul태그 내에 뿌려줌
+    $todoList.innerHTML = todoList.map(({ id, contents, completed, checked }) => (
+        `
+          <li data-id=${id} class=${completed}>
+              <div class="view">
+                  <input class="toggle" type="checkbox" ${checked}/>
+                  <label class="label">${contents}</label>
+                  <button class="destroy"></button>
+                  </div>
+            <input class="edit" value=${contents} />
+          </li>
+        `)
+      ).join('');
+      
+    // todo list의 총 item 갯수 표시
+    const length = todoList.length;
+    $todoCount.innerHTML = `총 <strong>${length}</strong> 개`;
+}
+
+// 사용자가 입력창에서 enter를 누를 때 동작
+$todoInput.addEventListener('keyup', ({ target, key }) => {
+    const contents = target.value;
+
+    // 눌린 키가 엔터가 아니거나 입력된 내용이 공백밖에 없으면 리턴
+    if (key !== "Enter" || !contents.trim()) return;
+
+    const id = newGuid(); // id 부여 (하단 newGuid() 함수 참고)
+    const completed = '';
+    const checked = '';
+    
+    todoList.push({ id, contents, completed, checked}); // 배열에 목록 정보 추가
+    target.value = ''; // 입력창 초기화
+
+    todoListRender(); // 목록 render
+})
+
+// 체크박스 클릭시 동작하는 이벤트 핸들러
+$todoList.addEventListener('click', ({target}) => {
+    // 클릭된 요소가 체크박스가 아니면 리턴
+    if (target.className !== 'toggle') return;
+    
+    // 가장 가까운 li 부모요소 찾기
+    const $li = target.closest('li');
+    const { id } = $li.dataset; // li요소에서 id 정보 가져오기
+
+    const idx = todoList.findIndex((todo) => todo.id === id); // 배열에서 해당 id를 가진 목록정보의 인덱스 찾아 idx에 저장
+    const completed = todoList[idx].completed; 
+    const checked = todoList[idx].checked;
+    
+    todoList[idx].completed = completed ? '' : 'completed';
+    todoList[idx].checked = checked ? '' : 'checked';
+    $li.className = completed ? '' : 'completed';
```

Comment:
```suggestion
    $li.classList.toggle("completed");
```

toggle 함수는 classList에 해당 클래스가 있으면 삭제하고 없으면 추가해주는 역할을 해줍니다!